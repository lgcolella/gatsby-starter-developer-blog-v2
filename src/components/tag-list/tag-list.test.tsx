/* Vendor imports */
import React from 'react'
import { mount } from 'enzyme'
import { Link } from 'gatsby'
/* App imports */
import TagList from './index'

describe('Tag List component', () => {
  test('matches snapshot', () => {
    let wrapper = mount(
      <TagList
        // Use duplicated and not sorted tags to test that component properly handles them
        tags={['nodejs', 'angular', 'laravel', 'electron', 'nodejs']}
      />
    )

    expect(wrapper).toMatchSnapshot()
  })

  test('properly filters, sorts and maps the passed tags', () => {
    let wrapper = mount(
      <TagList tags={['nodejs', 'angular', 'laravel', 'angular', 'nodejs']} />
    )

    expect(
      wrapper.find(Link).map(wrapperElement => wrapperElement.text())
    ).toEqual(['Angular', 'Laravel', 'Node.js'])
  })
})
