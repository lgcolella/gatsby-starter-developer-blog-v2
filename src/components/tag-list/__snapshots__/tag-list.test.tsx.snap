// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag List component matches snapshot 1`] = `
<TagList
  tags={
    Array [
      "nodejs",
      "angular",
      "laravel",
      "electron",
      "nodejs",
    ]
  }
>
  <div
    className="tags"
  >
    <mockConstructor
      key="angular"
      to="/tag/angular/"
    >
      <a
        href="/tag/angular/"
      >
        <Chip
          hoverEffect={true}
        >
          <label
            className="chip_with_hover "
          >
            Angular
          </label>
        </Chip>
      </a>
    </mockConstructor>
    <mockConstructor
      key="electron"
      to="/tag/electron/"
    >
      <a
        href="/tag/electron/"
      >
        <Chip
          hoverEffect={true}
        >
          <label
            className="chip_with_hover "
          >
            Electron
          </label>
        </Chip>
      </a>
    </mockConstructor>
    <mockConstructor
      key="laravel"
      to="/tag/laravel/"
    >
      <a
        href="/tag/laravel/"
      >
        <Chip
          hoverEffect={true}
        >
          <label
            className="chip_with_hover "
          >
            Laravel
          </label>
        </Chip>
      </a>
    </mockConstructor>
    <mockConstructor
      key="nodejs"
      to="/tag/nodejs/"
    >
      <a
        href="/tag/nodejs/"
      >
        <Chip
          hoverEffect={true}
        >
          <label
            className="chip_with_hover "
          >
            Node.js
          </label>
        </Chip>
      </a>
    </mockConstructor>
  </div>
</TagList>
`;
